lib/functional.js

```diff
@@ -19,4 +19,89 @@ module.exports = (api) => {
     value !== undefined && value !== null ? fn(value) : defVal
   );
 
+  // Either can be used for saving either left type or right type value.
+  // Left type is commonly used for additional information like error.
+  //
+  function Either(
+    // Either can be used for saving either left type or right type value.
+    // Left type is commonly used for additional information like error.
+    left, // left type value
+    right // right type value
+    // Returns tuple containing left type value or null as first element
+    // and right type value or undefined as second element
+  ) {
```

Comment:
@tshemsedinov, I am sorry it's my bad. I made if checking for a reason but forgot write good condition. I want that Either can save values of primitive types (not only errors) as left value. That wish needs complex checking of error.
```javascript
const [left, right] = api.common.makeEither(0); // left === 0, right === undefined
// Common checking
if (left) {
  console.log('left value');
} else {
  console.log('right value');
}
// prints 'right value'
// Complex checking
if (left !== null && left !== undefined) {
  console.log('left left');
} else {
  console.log('right value');
}
// prints 'left value'
```
That also shows hash implementation advantage. With tuple error checking has to be done every time. But with hash error checking is done once in constructor and then checking simplifies to checking `isLeft` flag